src/graphnet/models/graphs/graph_definition.py

```diff
@@ -447,3 +447,28 @@ def _add_custom_labels(
         for key, fn in custom_label_functions.items():
             graph[key] = fn(graph)
         return graph
+
+    def extra_repr_recursive(self, dictionary: dict, indent: int = 4) -> str:
+        """Recursively format a dictionary for extra_repr."""
+        result = "{\n"
+        for key, value in dictionary.items():
+            if key == "class_name":
+                continue
+            result += " " * indent + f"'{key}': "
+            if isinstance(value, dict):
+                result += self.extra_repr_recursive(value, indent + 4)
+            elif isinstance(value, Model):
+                result += value.__repr__()
+            else:
+                result += repr(value)
+            result += ",\n"
+        result += " " * (indent - 4) + "}"
+        return result
+
+    def extra_repr(self) -> str:
+        """Provide a more detailed description of the object print.
+
+        Returns:
+            str: A string representation containing detailed information about the object.
+        """
+        return f"{self.__class__.__name__}(\n{self.extra_repr_recursive(self._config.__dict__)})"
```

Comment:
I think perhaps it makes sense to move this into `Model`, that way we this more elaborate repr in most of our components. What are you thoughts?